packages/eas-cli/src/commands/device/delete.ts

```diff
@@ -3,251 +3,172 @@ import { Flags } from '@oclif/core';
+      const appleTeam = await selectAppleTeamOnAccountAsync({
+        accountName: account.name,
+        selectionPromptTitle: `What Apple Team would you like to list devices for?`,
+        paginatedQueryOptions,
+      });
+      appleTeamIdentifier = appleTeam.id;
+      appleTeamName = appleTeam.appleTeamName;
     }
 
     assert(appleTeamIdentifier, 'No team identifier is specified');
 
-    const appleDevicesResult = await this.getDevicesForTeamAsync(account.name, appleTeamIdentifier);
-
-    if (!appleDevicesResult) {
-      return;
-    }
-
-    const { appleTeamName, appleDevices } = appleDevicesResult;
+    const chosenDevice = udid
+      ? await AppleDeviceQuery.getByDeviceIdentifierAsync(account.name, udid)
+      : await selectAppleDeviceOnAppleTeamAsync({
+          accountName: account.name,
+          appleTeamIdentifier,
+          selectionPromptTitle: `Which device would you like to disable?`,
+          paginatedQueryOptions,
+        });
 
-    const chosenDevices = await this.chooseDevicesToDeleteAsync(appleDevices, udids);
+    this.logChosenDevice(chosenDevice, appleTeamName, appleTeamIdentifier, paginatedQueryOptions);
 
-    if (chosenDevices.length === 0) {
-      Log.newLine();
-      Log.warn('No devices were chosen to be removed.');
+    if (!(await this.shouldRemoveDeviceFromExpoAsync(paginatedQueryOptions))) {
       return;
     }
 
-    this.logChosenDevices(chosenDevices, appleTeamName, appleTeamIdentifier);
-
-    const hasRemoved = await this.askAndRemoveFromExpoAsync(chosenDevices);
+    await this.removeDeviceFromExpoAsync(chosenDevice);
 
-    if (!hasRemoved) {
-      return;
+    if (await this.shouldDisableDeviceOnAppleAsync(paginatedQueryOptions)) {
+      await this.disableDeviceOnAppleAsync(chosenDevice, appleTeamIdentifier);
     }
+  }
 
-    await this.askAndDisableOnAppleAsync(chosenDevices, appleTeamIdentifier);
+  async shouldDisableDeviceOnAppleAsync({
+    nonInteractive,
+  }: PaginatedQueryOptions): Promise<boolean> {
+    if (!nonInteractive) {
+      Log.newLine();
+      return await toggleConfirmAsync({
+        message: 'Do you want to disable this device on your Apple account as well?',
+      });
+    }
+    return true;
   }
 
-  async askAndDisableOnAppleAsync(
-    chosenDevices: (AppleDevice | AppleDeviceQueryResult)[],
+  async disableDeviceOnAppleAsync(
+    device: AppleDevice | AppleDeviceQueryResult,
     appleTeamIdentifier: string
   ): Promise<void> {
-    Log.newLine();
-    const deleteOnApple = await toggleConfirmAsync({
-      message: 'Do you want to disable these devices on your Apple account as well?',
-    });
-
-    if (!deleteOnApple) {
-      return;
-    }
-
     const ctx = await authenticateAsync({ teamId: appleTeamIdentifier });
     const context = getRequestContext(ctx);
 
     Log.addNewLineIfNone();
-    const removeAppleSpinner = ora('Disabling devices on Apple').start();
+    const removeAppleSpinner = ora('Disabling device on Apple').start();
```

Comment:
how does ora interact with json? do i need to use pretty print json output here or not print anything at all?